packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart

```diff
@@ -0,0 +1,116 @@
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import '../../camera_platform_interface.dart';
+
+/// The interface that implementations of camera must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `camera`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [CameraPlatform] methods.
+abstract class CameraPlatform extends PlatformInterface {
+  /// Constructs a CameraPlatform.
+  CameraPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static CameraPlatform _instance = MethodChannelCamera();
+
+  /// The default instance of [CameraPlatform] to use.
+  ///
+  /// Defaults to [MethodChannelCamera].
+  static CameraPlatform get instance => _instance;
+
+  /// Platform-specific plugins should set this with their own platform-specific
+  /// class that extends [CameraPlatform] when they register themselves.
+  static set instance(CameraPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Completes with a list of available cameras.
+  Future<List<CameraDescription>> availableCameras() {
+    throw UnimplementedError('availableCameras() is not implemented.');
+  }
+
+  /// Initializes the camera on the device and returns its cameraId.
+  Future<int> initializeCamera(
+    CameraDescription cameraDescription, {
+    ResolutionPreset resolutionPreset,
+    bool enableAudio,
+  }) {
+    throw UnimplementedError('initializeCamera() is not implemented.');
+  }
+
+  /// The camera's resolution has changed
+  Stream<ResolutionChangedEvent> onResolutionChanged(int cameraId) {
+    throw UnimplementedError('onResolutionChanged() is not implemented.');
+  }
+
+  /// The camera started to close.
+  Stream<CameraClosingEvent> onCameraClosing(int cameraId) {
+    throw UnimplementedError('onCameraClosing() is not implemented.');
+  }
+
+  /// The camera experienced an error.
+  Stream<CameraErrorEvent> onCameraError(int cameraId) {
+    throw UnimplementedError('onCameraError() is not implemented.');
+  }
+
+  /// Captures an image and returns the file where it was saved.
+  Future<XFile> takePicture(int cameraId) {
+    throw UnimplementedError('takePicture() is not implemented.');
+  }
+
+  /// Prepare the capture session for video recording.
+  Future<void> prepareForVideoRecording() {
+    throw UnimplementedError('prepareForVideoRecording() is not implemented.');
+  }
+
+  /// Starts a video recording and returns the file where it will be saved.
+  ///
+  /// The file is written on the flight as the video is being recorded.
+  /// The file can be read as soon as [stopVideoRecording] returns.
+  Future<XFile> startVideoRecording(int cameraId) {
```

Comment:
The file can be read as soon as [stopVideoRecording] returns. Should stopVideoRecording return the XFile instead? (It doesn't really matter much, but since this is a Future<XFile>, it's a little bit more clunky to use. For example, I don't think we could use `await` with this future, because that'd probably block the program (so stopVideoRecording would never be called))

Reply:
Having `stopVideoRecording` return the XFile indeed sounds like the more logical approach to me. 

The only case I could argue for `startVideoRecording` to return the XFile as well, is if there is any reason for users to access the file before stopping the recording. Currently I cannot seem to come up with any though.

Reply:
> The only case I could argue for startVideoRecording to return the XFile as well, is if there is any reason for users to access the file before stopping the recording. Currently I cannot seem to come up with any though.

I think the case where we'd want to be able to have a future returned from startVideoRecording is if we wanted to use some sort of UI widget, with a FutureBuilder or similar; but in the example app that is not done at all.